### ISSUE TITLE
Rename "Line" and related usages to FunctionOrFileScope?

### ISSUE THREAD

**Issue by arichard4** [lunarmodules/luacheck#35]
I found this extremely confusing when looking through the code- many usages of the word "line", and in particular the class "Line" from linearize.lua, are actually referring to a function or file scope, not a single line.

Unfortunately this isn't the same as a scope in general, as it doesn't create a new "Line" class for e.g. stepping inside an if branch; it's mostly just one scope/"Line" class per function, but with an additional one for the file's scope.

Does this sound right to ya'll? I'm not very familiar with the luacheck code, so I might be misunderstanding.

**Comment by arichard4** [comment]
Also, I'm curious if there's a better/shorter/clearer name for this. "scope" and "block" aren't quite right.

**Comment by alerque** [comment]
Is "closure" the word you're looking for here?

**Comment by arichard4** [comment]
Yes!

**Comment by alerque** [comment]
Yes, I think using the correct terminology here would be a good idea. Even for somebody unfamiliar with the term 'closure', using the *wrong* term 'line' just because it happens to be a familiar word isn't helping anybody, only making things more confusing. Better to have to look up the term and then understand what is going on than be mislead by our usage.
